tests/console/pam.pm

```diff
@@ -0,0 +1,60 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: This PM is to run a suite of tests about PAM, and gets a TAP
+#          format result. Four files(pam.sh, pam_test, pam_test.sh,
+#          system-default) are included. pam.sh is a "bats" script, and
+#          includes 25 tests for PAM. The other files are from the project
+#          "pam_test" which can be used to test a PAM stack for authentication
+#          and password change. The link: https://github.com/pbrezina/pam-test
+#   Steps:
+#       - add qa-head repo and install the tool "bats" from qa-head repo.
+#         Bats is a TAP-compliant testing framework for Bash.
+#       - download "pam_test" dir(automated tests of PAM) from "data" dir
+#         on openqa host if "prepare_test_data" perl module desn't run.
+#       - run the suite of tests in SUT.
+# Maintainer: Jun Wang <jgwang@suse.com>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use repo_tools 'add_qa_head_repo';
+
+sub run {
+    select_console 'root-console';
+
+    # bats is from qa repo, so adds qa repo and install bats
+    if (get_var('QA_HEAD_REPO')) {
+        add_qa_head_repo();
+    } else {
+        (my $version, my $service_pack) = split('-', get_required_var('VERSION'));
+        my $qa_head_repo = "http://download.suse.de/ibs/QA:/SLE$version$service_pack/update/";
+        zypper_ar("$qa_head_repo", name => 'qa-head');
+    }
+    zypper_call('install bats pam-config pam');
+
+    my $pamdir = "/home/bernhard/data/pam_test";
+    if (script_run("test -d /home/bernhard/data/pam_test"))
+    {
+        my $archive = "pam-tests.data";
+        assert_script_run(
+            sprintf "cd; curl -L -v %s/data/pam_test > %s && cpio -id < %s && mv data pam_test && rm -f %s",
```

Comment:
Please don't use sprintf, this makes the code a little bit harder to follow.

```suggestion
            "cd; curl -L -v ". autoinst_url . "/data/pam_test > $archive && cpio -id < $archive && mv data pam_test && rm -f $archive",
```

Reply:
I will improve it with other problems.